### PR TITLE
Port to Reality Management API

### PR DESCRIPTION
**Pull request by mcote01**
- port to Reality Management API
- updated tests
- marked `getRealityDataProjects` and added `getRealityDataITwins` to replace
- use URL type to manage urls
- updated changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ This log was last generated on Wed, 31 May 2023 21:24:48 GMT and should not be m
 
 <!-- Start content -->
 
+## 1.0.1 (2023-06-27)
```
Comment (aruniverse) [inline CHANGELOG.md:7]: the changelog would get auto-generated/updated by the changelog file you created.
everything you mentioned here should be included in the chnage/*.json files

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@itwin/reality-data-client",
-  "version": "1.0.0",
+  "version": "1.0.1",
```
Comment (aruniverse) [inline package.json:3]: version management, managed by the beachball changes

File: src/RealityData.ts [hunk 3]
```diff
@@ -82,7 +82,7 @@ export class ITwinRealityData implements RealityData {
   public classification?: string;
   public type?: string;
   public extent?: Extent;
-  public accessControl?: string;
+  public accessControl?: string; // DEPRECATED: not used in Reality Management API
```
Comment (aruniverse) [inline src/RealityData.ts:85]: might want to add a todo comment to remove in next major

File: src/RealityData.ts [hunk 4]
```diff
@@ -181,20 +181,23 @@ export class ITwinRealityData implements RealityData {
...
+        const url = new URL(`${this.client.baseUrl}/${this.id}/${ writeAccess === true ? "writeAccess" : "readAccess"}`);
 
-        const url = this.iTwinId ? `${this.client.baseUrl}/${this.id}/container/?projectId=${this.iTwinId}&access=${access}`
-          : `${this.client.baseUrl}/${this.id}/container/?&access=${access}`;
-        const requestOptions = getRequestConfig(accessTokenResolved, "GET", url, this.client.apiVersion);
+        if(this.iTwinId)
+          url.searchParams.append("iTwinId", this.iTwinId);
 
-        const response = await axios.get(url, requestOptions);
+        const requestOptions = getRequestConfig(accessTokenResolved, "GET", url.href, this.client.apiVersion);
```
Comment (aruniverse) [inline src/RealityData.ts:190]: where does the version come from?

File: src/RealityDataClient.ts [hunk 5]
```diff
@@ -207,14 +207,17 @@ export class RealityDataAccessClient implements RealityDataAccess {
 
   private extractContinuationToken(url: string | undefined): string | undefined {
     if (url) {
-      const continuationToken = url.split("&continuationToken=");
-      return continuationToken[continuationToken.length - 1];
+      const pattern = /continuationtoken=[a-zA-Z0-9]*/i;
```
Comment (matmarchand) [inline src/RealityDataClient.ts:210]: ContinuationToken are opaque to the caller and we should not assume which char they may contain. Using URLSearchParams would be more robust. https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/get#examples

File: src/RealityData.ts (hunk 4 above)
Comment (mcote01) [reply src/RealityData.ts:190]: right now it's an enum with one value : `ApiVersion.v1` which is also default.
it's used in the `getRequestConfig` function in `RequestOption.ts` to set up the "accept":"application/vnd.bentley.itwin-platform.v1+json" header that is required.

File: package.json (hunk 2 above)
Comment (mcote01) [reply package.json:3]: do you mean I must not change this line?

Comment (aruniverse) [reply package.json:3]: correct; take a look at:
https://github.com/iTwin/reality-data-client/blob/master/CONTRIBUTING.md#developer-notes (last bullet point)
https://github.com/microsoft/beachball

**Comment by mcote01** [comment]
I think I've covered everything I can think of, for 1.0.1. 
I will publish this PR when I'm back from vacation around July 10th.

File: src/RealityDataClient.ts [hunk 6]
```diff
@@ -70,7 +72,7 @@ export interface RealityDataResponse {
  */
 export class RealityDataAccessClient implements RealityDataAccess {
 
-  public readonly baseUrl: string = "https://api.bentley.com/realitydata";
+  public readonly baseUrl: string = "https://api.bentley.com/reality-management/reality-data";
```
Comment (aruniverse) [inline src/RealityDataClient.ts:75]: did the scopes change?

Comment (mcote01) [reply src/RealityDataClient.ts:75]: scopes are the same `realitydata:read`  `realitydata:modify`

**Pull request merged**